averaging/disc_aver_sw_paradiag_with_switches.py

```diff
@@ -193,28 +209,13 @@
     'aaos_jacobian_state': 'linear',
 }
 
-# setup parameters for paradiag solve (the number of windows is fixed to 1)
 if args.advection:
-    solver_parameters_diag['circulant_block_'+str(i)+'_'] = monoparameters_ns
-    print('set monoparameters_ns as solver_parameters_diag')
+    block_parameters = monoparameters_ns
 else:
-    solver_parameters_diag['circulant_block_'+str(i)+'_'] = hparams
-    print('set hparams as solver_parameters_diag')
-
-PETSc.Sys.Print('### === --- Calculating parallel solution --- === ###')
+    block_parameters = hparams
 
-# setup slice length
-slice_length_t = int(nt/args.nslices)
-assert(slice_length_t*args.nslices == nt)
-slice_length_t_half = int(nt/2/args.nslices)
-assert(slice_length_t_half*args.nslices == nt/2)
-slice_length_s = int(ns/args.nslices)
-assert(slice_length_s*args.nslices == ns)
-
-# setup time_partition
-time_partition_t = [slice_length_t for _ in range(args.nslices)]
-time_partition_t_half = [slice_length_t_half for _ in range(args.nslices)]
-time_partition_s = [slice_length_s for _ in range(args.nslices)]
+for i in range(sum(time_partition_t)):
+    solver_parameters_diag['circulant_block_'+str(i)+'_'] = block_parameters
```

Comment:
We found out that this approach isn't scalable as the window length increases because the options on each rank contains a ton of options is doesn't need for blocks on other ranks.
Instead you can just set default options for all blocks:
```suggestion
solver_parameters_diag['circulant_block'] = block_parameters
```
If you want to have different options for different blocks, then it's better to only set them for the blocks on the local ensemble member.

You can also mix default options with block-specific ones, e.g. you could set the default options then have:
```python
for i in local_block_indices:
    solver_parameters_diag[f'circulant_block_{i}_varying_option'] = varying_options[i]
```